### PR TITLE
Add requirement that per-try timeouts should have clear and actionable errors

### PR DESCRIPTION
**Pull request by bterlson**
@JeffreyRichter raised this point in a meeting the other day, I think it's important enough to lift to a guideline. What do you think?

### CODE REVIEW

File: docs/design/general/AzureCorePipeline.mdk [hunk 1]
```diff
@@ -117,6 +117,10 @@ retry if the service successfully responds indicating that it is throttling requ
 retry if the service responds with a 400-level response code unless a retry-after header is also returned.
 ~
 
+~ Must {#azurecore-retry-clear-errors}
+indicate when errors are thrown as a result of a retry policy's per-try timeout being exceeded, and include a note about how it should be configured by the user.
```
Comment (adrianhall) [inline docs/design/general/AzureCorePipeline.mdk]: Pre: MUST indicate when errors are thrown as a result of a retry policy's per-try timeout being exceeded, and include a note about how it should be configured by the user.

Post: SHOULD include information about relevant configuration when the HTTP pipeline throws an error.  

Not all errors will result in a configuration that can be set.

Comment (bterlson) [reply docs/design/general/AzureCorePipeline.mdk]: Seems fine to me!

Comment (bterlson) [reply docs/design/general/AzureCorePipeline.mdk]: Can add, "if there is a relevant configuration setting that would alleviate the problem"

Comment (adrianhall) [reply docs/design/general/AzureCorePipeline.mdk]: BOARD DECISION: APPROVED TO MERGE (After Changes)

**Comment by adrianhall** [comment]
Integrated into PR #537 (Jekyll rebuild)

**Pull request closed without merging**